acvm/src/compiler.rs

```diff
@@ -68,23 +68,28 @@ pub fn compile(
     let mut transformed_gates = Vec::new();
 
     let mut next_witness_index = acir.current_witness_index + 1;
+    let mut intermediate_variables: IndexMap<Expression, (FieldElement, Witness)> = IndexMap::new();
```

Comment:
I think it would be good to document what this map is supposed to contain. Finding out what this `FieldElement` is requires you to trace through several functions.